### ISSUE TITLE
Remove for comprehensions in RTS.scala

### ISSUE THREAD

**Issue by jdegoes** [scalaz/scalaz-zio#73]
For comprehensions in `RTS.scala` add a useless final `map` operation, which slows down performance. They can all be eliminated by rewriting in terms of `flatMap`.

**Comment by ktonga** [comment]
Or... we could also enable this plugin https://github.com/oleg-py/better-monadic-for

**Comment by jdegoes** [comment]
I'd rather keep the project as small as possible. We need to support Dotty too.

**Comment by tuleism** [comment]
Currently, I only see one use of for comprehension in `RTS.scala`:

https://github.com/scalaz/scalaz-zio/blob/98559920c984545832f4c9bfa339c67fa4204f4b/core/jvm/src/main/scala/scalaz/zio/RTS.scala#L567-L571

It seems the final `map` operation is needed here, or we would need to rewrite `exitUninterruptible`. Am I missing something?

**Comment by tuleism** [comment]
For `enterUninterruptible *> io.io`, it's clear to me that the final `map` will disappear compare to using `for` comprehension.

But for `io.io <* exitUninterruptible`, I still see the `map` because it is translated to:

```scala
io.io.flatMap(v => exitUninterruptible.map(_ => v))
```
Or are you saying that with `for` comprehension, we have one additional `map`?

**Comment by ktonga** [comment]
Yeah, you right, the `map` is still there. But I don't think we can get rid of it in this case

**Comment by ktonga** [comment]
In this case there is no difference, this optimization is only valid when the `for` yields the result of the last `flatMap`

**Comment by tuleism** [comment]
@ktonga thanks!
